### PR TITLE
Polishing README

### PR DESCRIPTION
**Pull request by mgr0dzicki**

### CODE REVIEW

**Review by obi1kenobi — APPROVED** [review]
This is certainly an improvement, and I'm happy to merge it.

There's further room for improvement. Consider a user coming to this page. They probably have a _specific_ question in mind, like "how do I exclude a package from being checked" for example. Currently, none of the entries in the table of contents are a direct match, and they'd have to navigate indirectly by going to the "use with more than one crate" and then reading through a bunch of text.

The way we can do better is by splitting up those large sections into smaller ones that target a very specific use case that users are likely to want, and make it possible for a new user to come to the page, find an entry that exactly matches what they are trying to do, and find an example that they can directly copy-paste into their own project without needing to dig too much through the docs.

In other words, we're looking to minimize "time to solution" for our users, for all the most common questions and problems that they might come up with.

**Comment by mgr0dzicki** [comment]
You're right, I feel the README seems OK if it is read as a whole, but finding specific examples in it might be tedious... I have to go and won't be able to work on it until tomorrow. Can we wait with releasing the action?

**Comment by obi1kenobi** [comment]
Let's just merge this and release on schedule as we planned. People will always see the README on the current `main` branch anyway, so nothing is stopping us from continuing to improve it. This PR already makes it better and it'll be even better still with the next pass.
